src/mesh/mesh.cpp

```diff
@@ -1115,18 +1144,23 @@ void Mesh::Initialize(bool init_problem, ParameterInput *pin, ApplicationInput *
     }
 
     //  Now do prolongation, compute primitives, apply BCs
-    for (int i = 0; i < nmb; ++i) {
-      auto &mbd = block_list[i]->meshblock_data.Get();
-      ApplyBoundaryConditions(mbd);
-      // Call MeshBlockData based FillDerived functions
-      Update::FillDerived(mbd.get());
-    }
     for (int i = 0; i < num_partitions; i++) {
       auto &md = mesh_data.GetOrAdd("base", i);
+      if (multilevel) {
+        ApplyCoarseBoundaryConditions(md);
+        ProlongateBoundaries(md);
+      }
+      ApplyFineBoundaryConditions(md);
       // Call MeshData based FillDerived functions
       Update::FillDerived(md.get());
     }
 
+    for (int i = 0; i < nmb; ++i) {
+      auto &mbd = block_list[i]->meshblock_data.Get();
+      // Call MeshBlockData based FillDerived functions
+      Update::FillDerived(mbd.get());
```

Comment:
Potential merge conflict with my precomm fill derived PR. Just need to make sure git doesn't put the precomm stuff below the boundaries/comms

Reply:
It looks like it merged to the correct place